app/src/main/java/com/example/weather/ui/forecast/ForecastViewModel.java

```diff
@@ -27,21 +24,12 @@
         return liveData;
     }
 
-    private LiveData<List<WeatherData>> fetchAllSavedWeather(){
+    public LiveData<List<WeatherData>> fetchAllSavedWeather(){
         return RepositoryImpl.getInstance().getAllWeather();
     }
 
     @SuppressLint("CheckResult")
-    public void getCurrentWeatherByCoord(double lat, double lon, TextView textView){
-        RepositoryImpl.getInstance().getCurrentWeatherDataByCoord(""+lat,""+lon, "metric")
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(new Consumer<Float>() {
-                    @SuppressLint("SetTextI18n")
-                    @Override
-                    public void accept(Float aFloat) throws Exception {
-                        textView.setText(""+aFloat);
-                    }
-                });
+    public Observable<Float> getCurrentWeatherByCoord(double lat, double lon){
```

Comment:
Previous version was right, with the exception of updating TextView inside ViewModel.